src/session/session_api.c

```diff
@@ -267,7 +267,8 @@ __session_close(WT_SESSION *wt_session, const char *config)
     SESSION_API_CALL_PREPARE_ALLOWED(session, close, config, cfg);
     WT_UNUSED(cfg);
 
-    return (__wt_session_close_internal(session));
+    WT_ERR(__wt_session_close_internal(session));
```

Comment:
If this call fails, it was skipping the `API_END_RET_NOTFOUND_MAP` even though the session wasn't freed.